lib/resource-mapper.js

```diff
@@ -29,16 +38,28 @@ class ResourceMapper {
   }
 
   // Maps the request for a given resource and representation format to a server file
+  // When the URL ends with a '/', then files with the prefix 'index.' will be matched,
+  // such as 'index.html' and 'index.ttl'.
   async mapUrlToFile ({ url, contentType, createIfNotExists }) {
-    const fullPath = this._getFullPath(url)
+    let fullPath = this._getFullPath(url)
+    let isIndex = fullPath.endsWith('/')
     let path
 
+    // Append index filename if the URL ends with a '/'
+    if (isIndex) {
+      fullPath += this._indexName
+    }
+
     // Create the path for a new file
     if (createIfNotExists) {
       path = fullPath
       // If the extension is not correct for the content type, append the correct extension
       if (this._getContentTypeByExtension(path) !== contentType) {
-        path += contentType in extensions ? `$.${extensions[contentType][0]}` : '$.unknown'
+        // Append a '$', unless we map for the index
+        if (!isIndex) {
+          path += '$'
+        }
```

Comment:
I don't get this change I'm afraid. Are there situations where we end up with multiple dollar signs now?

Reply:
> Are there situations where we end up with multiple dollar signs now?

No, at most 1 dollar sign will be present.

> I don't get this change I'm afraid.

This is needed for this situation:
PUT on `http://localhost/space/` with content-type `text/html` should create `/space/index.html`, instead of `/space/index$.html`.
(Unless we don't want this?...)